crates/net/network/src/network.rs

```diff
@@ -205,8 +205,10 @@ impl NetworkHandle {
     }
 
     /// Send message to gracefully shutdown node.
-    pub fn shutdown(&self) {
-        self.send_message(NetworkHandleMessage::Shutdown);
+    pub async fn shutdown(&self) -> Result<(), oneshot::error::RecvError> {
```

Comment:
this should include a brief explainer what it actually does:
- disconnect all session
- prevent connections